articles/116_legacy_interface_definition.md

```diff
@@ -144,7 +144,7 @@ Additional rule for `string` arrays:
 
 ### Constants
 
-Constants are defined by a *primitive type*, a *name* as well as a *fixed value*.
+Constants are defined by a *primitive type*, a *name* as well as a *fixed value*. The *fixed value* can be specified in binary, octal, decimal, or hex; e.g. 0b01, 0o01, 01, 0x01.
```

Comment:
The idea of this change is reasonable to me.  However, there are a few issues with how it is presented here:

1.  This change specifies the way that Python encodes integers in the various bases.  I like that change, as Python is  explicit about the prefix (`0b`, `0o`, or `0x`).  However, that is not the only way to do it, so I think the language needs to be a little stronger than just an `e.g.` here.  We should be a bit more explicit about what we are supporting, maybe by changing this to a list of what is supported.
2. I'm pretty sure that Python disallows a leading 0, so the integer example should change.

Reply:
> I'm pretty sure that Python disallows a leading 0, so the integer example should change.

Just clarifying that only `01` is invalid in python 3.6.9

```python
>>> 0b01
1
>>> 0o01
1
>>> 01
  File "<stdin>", line 1
    01
     ^
SyntaxError: invalid token
>>> 0x01
1
```

Reply:
> Just clarifying that only `01` is invalid in python 3.6.9

Oh!  Yes, good call, that is exactly what I meant.  Obviously a leading 0 is allowed with a secondary prefix, like `0x`, `0b`, or `0o`.

Reply:
Oops, to add to the confusion - `int()` in python 3 allows the leading 0

```python
>>> int('01')
1
```

So the implementation in ros2/rosidl#559 would allow `01` :upside_down_face:

Reply:
Not with a base 0, though:

```
>>> int('01')
1
>>> int('01', 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 0: '01'
```

Reply:
Also, if we are going to update this specification to be Python compliant, we should go all the way and just make sure it supports everything as listed in the documentation: https://docs.python.org/3/library/functions.html#int